xds/src/main/java/io/grpc/xds/XdsClientMetricReporterImpl.java

```diff
@@ -157,19 +163,27 @@ void reportCallbackMetrics(BatchRecorder recorder, XdsClient xdsClient) {
 
   private void computeAndReportResourceCounts(
       Map<XdsResourceType<?>, Map<String, ResourceMetadata>> metadataByType,
+      Map<XdsResourceType<?>, Map<String, String>> authorityByType,
       MetricReporterCallback callback) {
     for (Map.Entry<XdsResourceType<?>, Map<String, ResourceMetadata>> metadataByTypeEntry :
         metadataByType.entrySet()) {
       XdsResourceType<?> type = metadataByTypeEntry.getKey();
 
       Map<String, Long> resourceCountsByState = new HashMap<>();
-      for (ResourceMetadata metadata : metadataByTypeEntry.getValue().values()) {
+      Map<String, String> authorityByState = new HashMap<>();
+      for (Map.Entry<String, ResourceMetadata> metadataByName :
+              metadataByTypeEntry.getValue().entrySet()) {
+        String resourceName = metadataByName.getKey();
+        ResourceMetadata metadata = metadataByName.getValue();
         String cacheState = cacheStateFromResourceStatus(metadata.getStatus(), metadata.isCached());
         resourceCountsByState.compute(cacheState, (k, v) -> (v == null) ? 1 : v + 1);
+        authorityByState.put(cacheState, authorityByType.get(type).get(resourceName));
```

Comment:
This just randomly assigns authorities to cache states. Authority is another dimension.